src/main/java/appeng/spatial/StorageCellLot.java

```diff
@@ -0,0 +1,85 @@
+package appeng.spatial;
+
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * A lot inside the storage cell world that is assigned to a specific storage cell.
+ */
+public class StorageCellLot {
+
+    private static final int MAX_SIZE = 512;
+
+    /**
+     * Id of the lot.
+     */
+    private final int id;
+
+    /**
+     * The storage size of this dimension. This is dicateted by the pylon structure
+     * size used to perform the first transfer into this dimension. Once it's set,
+     * it cannot be changed anymore.
+     */
+    private final BlockPos size;
+
+    /**
+     * AE2 player id of who primarily owned the network when the storage lot was
+     * allocated.
+     */
+    private final int owner;
+
+    public StorageCellLot(int id, BlockPos size, int owner) {
+        this.id = id;
+        this.size = size;
+        this.owner = owner;
+        if (size.getX() < 1 || size.getY() < 1 || size.getZ() < 1) {
+            throw new IllegalArgumentException("Lot size " + size + " is smaller than minimum size.");
+        }
+        if (size.getX() > MAX_SIZE || size.getY() >= MAX_SIZE || size.getZ() >= MAX_SIZE) {
+            throw new IllegalArgumentException("Lot size " + size + " exceeds maximum size of " + MAX_SIZE);
+        }
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public BlockPos getSize() {
+        return size;
+    }
+
+    public int getOwner() {
+        return owner;
+    }
+
+    public BlockPos getOrigin() {
```

Comment:
This probably needs some documentation on how it layouts the lots. This appears to at least try and use all 4 direction, which is pretty much a requirement. As far as I am aware, minecraft still uses floats for positions and at some point the start to get into rounding errors. This is usually not noticable, as players won't easily reach coordinates in the range of 100k or even more. But with cells someone could just loop through 1k cells and end up half a million blocks away from the world origin. Which might start to break down physics like collisions.